apps/meteor/client/components/UserAutoCompleteMultiple/UserAutoCompleteMultipleFederated.tsx

```diff
@@ -55,33 +55,55 @@ const UserAutoCompleteMultipleFederated = ({
 		{ keepPreviousData: true },
 	);
 
-	const options = data || [];
+	const options = useMemo(() => data || [], [data]);
 
-	const onAddSelected: ComponentProps<typeof Options>['onSelect'] = ([value]) => {
-		setFilter('');
-		const cachedOption = options.find(([curVal]) => curVal === value)?.[1];
-		if (!cachedOption) {
-			throw new Error('UserAutoCompleteMultiple - onAddSelected - failed to cache option');
-		}
-		setSelectedCache({ ...selectedCache, [value]: cachedOption });
-	};
+	const onAddSelected = useCallback(
+		([value]: OptionType, shouldRemoveUser = false): void => {
+			setFilter('');
+			if (shouldRemoveUser) {
```

Comment:
This doesn't really make sense inside this function. Instead of passing an additional prop, I suggest making a new function for removing from cache.

Reply:
Yes, actually this function was earlier used at other places, so followed this pattern, forgot to remove it.